includes/components/class-admin.php

```diff
@@ -102,11 +105,144 @@ public function __construct( $args = [] ) {
 
 			// Render footer.
 			add_action( 'admin_footer', [ $this, 'render_footer' ] );
+
+            // Disable extensions.
+            add_action( 'admin_init', [ $this, 'disable_extensions' ] );
 		}
 
 		parent::__construct( $args );
 	}
 
+    /**
+     * Updates license key option.
+     */
+    public function update_license_key() {
+
+        // Clear purchased extensions cache.
+        hivepress()->cache->delete_cache( 'purchased_plugins' );
+
+        // Clear purchased themes cache.
+        hivepress()->cache->delete_cache( 'purchased_themes' );
+    }
+
+    /**
+     * Disables unlicensed extensions.
+     */
+    public function disable_extensions() {
+        require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+        foreach ( $this->get_unlicensed_products()['extensions'] as $extension ) {
+            if ( ! is_plugin_active( $extension['file'] ) ) {
+                continue;
+            }
+
+            // Deactivate extension.
+            deactivate_plugins( $extension['file'] );
+        }
+    }
+
+    /**
+     * Gets installed unlicensed products.
+     *
+     * @return array
+     */
+    public function get_unlicensed_products() {
```

Comment:
Avoid public helpers because once we release this we can't later deprecate the helper since some developers may use it from the outside code. You can use protected in this case